### PR TITLE
Modernize the readme's Status paragraph

### LINKED ISSUE #9127: The "Status" paragraph in the readme seems very outdated
### Apache Iceberg version

None

### Query engine

None

### Please describe the bug 🐞

Looking at the ReadMe in the root of this repository, one would be mistaken for thinking we're living in Table v1 times.

**Quote start:**

Iceberg is under active development at the Apache Software Foundation.

The core Java library that tracks table snapshots and metadata is complete, but still evolving. Current work is focused on adding row-level deletes and upserts, and integration work with new engines like Flink and Hive.

The [Iceberg format specification](https://iceberg.apache.org/spec) is being actively updated and is open for comment. Until the specification is complete and released, it carries no compatibility guarantees. The spec is currently evolving as the Java reference implementation changes.

[Java API javadocs](https://iceberg.apache.org/javadoc/latest) are available for the main.

**Quote end**

That last sentence seems incomplete. But more importantly, I think the information provided is, today at least, no longer correct. I've been told there _are_ compatibility guarantees, and what's being worked on today is probably no longer row-level deletes and upserts.

I would like to suggest that an update, or perhaps a slimming down of the text, is overdue.

### PR DESCRIPTION
**Pull request by mt-ronkorving**
I've updated the ReadMe's Status paragraph to be more inline with the reality today and hopefully won't have to be kept up-to-date going forward. Please let me know if there are any inaccuracies in the text, or stylistic changes anyone would like to see. Thank you.

Closes #9127

### CODE REVIEW

**Review by nastra — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -31,15 +31,17 @@ Background and documentation is available at <https://iceberg.apache.org>
 
 Iceberg is under active development at the Apache Software Foundation.
 
-The core Java library that tracks table snapshots and metadata is complete, but still evolving. Current work is focused on adding row-level deletes and upserts, and integration work with new engines like Flink and Hive.
+The [Iceberg format specification][iceberg-spec] is considered stable.
```
Comment (bitsondatadev) [inline README.md:34]:
```suggestion
The [Iceberg format specification][iceberg-spec] is stable.
```

File: README.md [hunk 2]
```diff
@@ -31,15 +31,17 @@ Background and documentation is available at <https://iceberg.apache.org>
...
 
-The [Iceberg format specification][iceberg-spec] is being actively updated and is open for comment. Until the specification is complete and released, it carries no compatibility guarantees. The spec is currently evolving as the Java reference implementation changes.
+The core Java library can be found in this repository and is considered the reference implementation for other libraries.
```
Comment (bitsondatadev) [inline README.md:36]:
```suggestion
The core Java library is located in this repository and is the reference implementation for other libraries.
```

**Review by bitsondatadev — APPROVED** [review]
Few small nits. This is much better!! Thanks @mt-ronkorving !!

**Comment by mt-ronkorving** [comment]
@bitsondatadev Thanks for the suggestions, I've merged them in

File: README.md [hunk 3]
```diff
@@ -31,15 +31,17 @@ Background and documentation is available at <https://iceberg.apache.org>
 
 Iceberg is under active development at the Apache Software Foundation.
 
-The core Java library that tracks table snapshots and metadata is complete, but still evolving. Current work is focused on adding row-level deletes and upserts, and integration work with new engines like Flink and Hive.
+The [Iceberg format specification][iceberg-spec] is stable.
```
Comment (szehon-ho) [inline README.md:34]: I think its good, but we will have more versions like v2, v3.  Not sure if its worth mentioning like 'is versioned'?

Comment (Fokko) [reply README.md:34]: That's a good suggestion @szehon-ho, what do you think of:

```suggestion
The [Iceberg format specification][iceberg-spec] is stable and new features are added with each version.
```

Comment (szehon-ho) [reply README.md:34]: Yes, great!  It's what i was thinking but couldnt find the words!

Comment (Fokko) [reply README.md]: Nice, thanks for letting us know! 👍

**Comment by Fokko** [comment]
Thanks for taking care of this @mt-ronkorving and thanks for the reviews @bitsondatadev, @nastra and @szehon-ho 🙌

**Pull request merged**